### ISSUE TITLE
bottled `ncdu` cannot run on linux

### ISSUE THREAD

**Issue by ycrack** [Homebrew/homebrew-core#92282]
### `brew gist-logs <formula>` link OR `brew config` AND `brew doctor` output

```shell
# brew config
HOMEBREW_VERSION: 3.3.9
ORIGIN: https://github.com/Homebrew/brew
HEAD: 96137bc19e68398ebbb7033379df288cd8b9a3f9
Last commit: 13 days ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 49611bbc2318e2b0d2f49e170210bc9c93fe36c7
Core tap last commit: 2 hours ago
Core tap branch: master
HOMEBREW_PREFIX: /home/linuxbrew/.linuxbrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_CORE_GIT_REMOTE: https://github.com/Homebrew/homebrew-core
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 2.6.8 => /home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.8/bin/ruby
CPU: octa-core 64-bit haswell
Clang: 13.0.0
Git: 2.25.1 => /bin/git
Curl: 7.68.0 => /bin/curl
Kernel: Linux 5.4.0-91-generic x86_64 GNU/Linux
OS: Ubuntu 20.04.3 LTS (focal)
Host glibc: 2.31
/usr/bin/gcc: 9.3.0
/usr/bin/ruby: N/A
glibc: N/A
gcc@5: N/A
xorg: N/A

# brew doctor
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Some installed formulae are deprecated or disabled.
You should find replacements for the following formulae:
  guile@2
```

### Verification

- [X] I ran `brew update` and am still able to reproduce my issue.
- [X] I have resolved all warnings from `brew doctor` and that did not fix my problem.

### What were you trying to do (and why)?

upgrade `ncdu` (from 1.16 to 2.0)

### What happened (include all command output)?

After `brew upgrade` done, the `ncdu` command fails with a message "illegal (hardware) instruction".
When I reinstall with `--build-from-source` flag, I can run the `ncdu` command correctly.

### What did you expect to happen?

`ncdu` works correctly

### Step-by-step reproduction instructions (by running `brew` commands)

```shell
brew upgrade ncdu
```

**Comment by carlocab** [comment]
Yes, it seems Zig compiles with the equivalent of `-march=native` by default. We need to fix the formula so that it passes the value of `Hardware.oldest_arch` instead.

**Comment by carlocab** [comment]
Should be fixed by #92321.
